diff --git a/android/src/debug/java/expo/modules/devlauncher/DevLauncherController.kt b/android/src/debug/java/expo/modules/devlauncher/DevLauncherController.kt
index a6b83588a9cd4280c4d36dfcccaac0167d11c6d3..85f5badd995aa312b4e05e28f58d628b47179201 100644
--- a/android/src/debug/java/expo/modules/devlauncher/DevLauncherController.kt
+++ b/android/src/debug/java/expo/modules/devlauncher/DevLauncherController.kt
@@ -317,6 +317,15 @@ class DevLauncherController private constructor(
   }

   override fun getCurrentReactActivityDelegate(activity: ReactActivity, delegateSupplierDevLauncher: DevLauncherReactActivityDelegateSupplier): ReactActivityDelegate {
+    if (
+      mode == Mode.APP &&
+      !appIsLoading &&
+      activity.intent?.action == Intent.ACTION_MAIN &&
+      appHost.currentReactContext?.currentActivity == null
+    ) {
+      mode = Mode.LAUNCHER
+    }
+
     return if (mode == Mode.LAUNCHER) {
       DevLauncherReactActivityRedirectDelegate(activity, this::redirectFromStartActivity)
     } else {
diff --git a/android/src/debug/java/expo/modules/devlauncher/launcher/DevLauncherRecentlyOpenedAppsRegistry.kt b/android/src/debug/java/expo/modules/devlauncher/launcher/DevLauncherRecentlyOpenedAppsRegistry.kt
index ad21086a1a65058ab54e8dff4232b652baebf1ba..53334c1ebb8b1381ec238a2bc636fb4c13944fad 100644
--- a/android/src/debug/java/expo/modules/devlauncher/launcher/DevLauncherRecentlyOpenedAppsRegistry.kt
+++ b/android/src/debug/java/expo/modules/devlauncher/launcher/DevLauncherRecentlyOpenedAppsRegistry.kt
@@ -90,13 +90,9 @@ class DevLauncherRecentlyOpenedAppsRegistry(context: Context) {
   }

   fun getMostRecentApp(): DevLauncherAppEntry? {
-    val recentlyOpenedApps = getRecentlyOpenedApps()
-
-    return if (recentlyOpenedApps.isNotEmpty()) {
-      recentlyOpenedApps.maxByOrNull { it.timestamp }
-    } else {
-      null
-    }
+    return getRecentlyOpenedApps()
+      .filter { it.isEASUpdate == true }
+      .maxByOrNull { it.timestamp }
   }

   fun clearRegistry() {
